### ISSUE TITLE
It would be great to be able to copy/move files between folders

### ISSUE THREAD

**Issue by valp124** [os-js/osjs-filemanager-application#7]

**Comment by andersevenrud** [comment]
I have a somewhat working clipboard solution ready, so I guess this would be a good place to test it out :blush:

**Comment by andersevenrud** [comment]
I've started work on this and opened the following pull requests:

* https://github.com/os-js/osjs-client/pull/79
* https://github.com/os-js/osjs-filemanager-application/pull/8

**Comment by andersevenrud** [comment]
FYI I actually found a bug in the copy/move functionality in the server. So I'd advice upgrading `@osjs/server` :)

**Comment by valp124** [comment]
Thanks for the heads up!!

**Comment by andersevenrud** [comment]
@valp124 @josephjeno 

If you update `@osjs/client` and `@osjs/filemanager-application` you should now be able to copy/cut and paste files via the context menu as well as a new `Edit` menubar item :)

**Comment by andersevenrud** [comment]
I also just pushed another update of `@osjs/filemanager-application` that adds in proper drop support.

**Comment by valp124** [comment]
What an awesome dude you are!!

**Comment by josephjeno** [comment]
@andersevenrud ah this is awesome!  

I've noticed when slightly moving a file and releasing, I receive an alert about source and destination being the same. It's just a minor annoyance, but can this alert be squashed? I am triggering it a lot just from touching files a bit, also maybe I'm too click-happy.

![image](https://user-images.githubusercontent.com/29079278/70479668-df853400-1aab-11ea-9eab-f410f4230559.png)
![image](https://user-images.githubusercontent.com/29079278/70479782-19eed100-1aac-11ea-91d8-077c787919fe.png)

**Comment by andersevenrud** [comment]
@josephjeno Aha. I'll add a check to prevent this :)

**Comment by andersevenrud** [comment]
@josephjeno Just published an update that takes care of this. Thanks!

**Comment by valp124** [comment]
Thank you so much, Anders, from both @josephjeno and me!!

And since we're on the subject ... :) No urgency at all, these would be bells and whistles, but it'd be super convenient to be able to do two more things:

1) Drag and drop files into a folder within the same window, as such

![image](https://user-images.githubusercontent.com/7969539/70553163-a058f080-1b48-11ea-86b0-7caaff36c613.png)

2) Drag and drop folders (in addition to files), either to other folders or to the desktop, as such

![image](https://user-images.githubusercontent.com/7969539/70553527-486eb980-1b49-11ea-9b23-6d606edc2e4f.png)

(I suppose, dragging to the desktop would then create a shortcut rather than a copy.)

Again, these are more ideas than requests, not critical at all!

**Comment by andersevenrud** [comment]
> Drag and drop files into a folder within the same window, as such

I'll look into it. The UI component should support this :thinking: 

> Drag and drop folders (in addition to files), either to other folders or to the desktop, as such

I'm not sure why I disabled dragging of folders.... but I've got a hunch that folder copy was not supported when I first added DnD. So I'll look into this again now that it most likely works just fine.

The desktop icons should support folders just fine as well since this is using the underlying VFS. I even see in the code that there's a conditional to open folders using File Manager :)

> (I suppose, dragging to the desktop would then create a shortcut rather than a copy.)

It's possible to hold <kbd>shift</kbd> when dropping to get a contextmenu for creating shortcuts instead of copy on the desktop. *However* I've seen some issues relating to this that I'll look into.

**Comment by andersevenrud** [comment]
> (I suppose, dragging to the desktop would then create a shortcut rather than a copy.)

But having folders as shortcuts by default is probably a good idea :+1:

**Comment by valp124** [comment]
Right, exactly, open File Manager when double-clicking on a folder! :blush:
